pkg/algorithm/cell_allocation.go

```diff
@@ -89,21 +90,59 @@ func getLowestFreeCellLevel(freeList ChainCellList, l CellLevel) CellLevel {
 		"even split to the highest level %v", l-1))
 }
 
+// when buddyAlloc allocates bad cells,
+// check whether it is safe to split a higher level cell to get current level cells.
+func checkSplitSafety(freeList ChainCellList, freeCellNum map[CellLevel]int32, l CellLevel) bool {
+	var splitableCell Cell
+	splitableNum := map[CellLevel]int32{}
+	for i := CellLevel(len(freeList)); i >= CellLevel(1); i-- {
+		// calculate splitable number
```

Comment:
>calculate splitable number [](start = 5, length = 26)

Seems we can early stop the "calculate splitable number" to l+1

Reply:
check safety to panic

Reply:
(Background: The panic will only fail current pod schedule, instead of exit the whole process)
Lower level safety problems seems does not necessary to fail current allocation, do we need to panic here? 
@zhypku any suggestion?

Reply:
Seems current behaiour is: even if there are safety broken, but if the broken does not impact current pod scheduling, the pod can still be scheduled.
If so, we can insist this behaiour.

Reply:
early stop at `current level + 1`